File: app/src/main/java/edu/tjrac/swant/todo/view/GroupWebViewerActivity.java
package edu.tjrac.swant.todo.view;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.view.ViewPager;

import com.yckj.baselib.common.adapter.FragmentsPagerAdapter;
import com.yckj.baselib.common.base.BaseActivity;
import com.yckj.baselib.common.base.BaseFragment;
import com.yckj.baselib.common.base.BaseWebViewFragment;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import edu.tjrac.swant.todo.adapter.GroupWebViewPageTransation;
import edu.tjrac.swant.todo.bean.WebInfo;
import edu.tjrac.swant.unicorn.R;

public class GroupWebViewerActivity extends BaseActivity {

    @BindView(R.id.viewpager) ViewPager mViewpager;

    FragmentsPagerAdapter adapter;
    List<WebInfo> infos;
BaseFragment current;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_group_web_viewer);
        ButterKnife.bind(this);
        infos = getIntent().getParcelableArrayListExtra("infos");

        adapter = new FragmentsPagerAdapter(getSupportFragmentManager());
        for (WebInfo item : infos) {
            adapter.addFragment(new BaseWebViewFragment(item.getUrl()
            ,R.layout.activity_service_text), item.getTitle());
        }
        mViewpager.setOffscreenPageLimit(infos.size());

        mViewpager.setAdapter(adapter);
        mViewpager.setPageMargin(40);
        mViewpager.setPageTransformer(true, new GroupWebViewPageTransation());
        mViewpager.setOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                current= (BaseFragment) adapter.getItem(position);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });

    }


    public static void start(Context context, ArrayList<WebInfo> infos) {
        context.startActivity(new Intent(context, GroupWebViewerActivity.class)
                .putExtra("infos", infos)
                .addFlags(Intent.FLAG_ACTIVITY_LAUNCH_ADJACENT | Intent.FLAG_ACTIVITY_NEW_TASK)
        );
    }

    @Override
    public void onBackPressed() {
        if(current.isBackable()){
            current.onBack();
            return;
        }
        super.onBackPressed();
    }
}

File: app/src/main/java/edu/tjrac/swant/todo/view/GroupWebViewerActivity.kt
package edu.tjrac.swant.todo.view

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.view.ViewPager
import com.yckj.baselib.common.adapter.FragmentsPagerAdapter
import com.yckj.baselib.common.base.BaseActivity
import com.yckj.baselib.common.base.BaseFragment
import com.yckj.baselib.common.base.BaseWebViewFragment
import edu.tjrac.swant.todo.adapter.GroupWebViewPageTransation
import edu.tjrac.swant.todo.bean.WebInfo
import edu.tjrac.swant.unicorn.R
import kotlinx.android.synthetic.main.activity_group_web_viewer.*
import java.util.*

class GroupWebViewerActivity : BaseActivity() {


    internal var adapter: FragmentsPagerAdapter?=null
    internal var infos: List<WebInfo>?=null
    internal var current: BaseFragment?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
                setContentView(R.layout.activity_group_web_viewer);
        infos = intent.getParcelableArrayListExtra<WebInfo>("infos")

        adapter = FragmentsPagerAdapter(supportFragmentManager)
        for (item in infos!!) {
            adapter!!.addFragment(BaseWebViewFragment(item.url, R.layout.activity_service_text), item.title)
        }
        viewpager.setOffscreenPageLimit(infos!!.size)

        viewpager.setAdapter(adapter)
        viewpager.setPageMargin(40)
        viewpager.setPageTransformer(true, GroupWebViewPageTransation())
        viewpager.setOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

            }

            override fun onPageSelected(position: Int) {
                current = adapter!!.getItem(position) as BaseFragment
            }

            override fun onPageScrollStateChanged(state: Int) {

            }
        })

    }

    override fun onBackPressed() {
        if (current!!.isBackable) {
            current!!.onBack()
            return
        }
        super.onBackPressed()
    }

    companion object {


        fun start(context: Context, infos: ArrayList<WebInfo>) {
            context.startActivity(Intent(context, GroupWebViewerActivity::class.java)
                    .putExtra("infos", infos)
                    .addFlags(Intent.FLAG_ACTIVITY_LAUNCH_ADJACENT or Intent.FLAG_ACTIVITY_NEW_TASK)
            )
        }
    }
}
